addons/trino-extractor/src/main/java/org/apache/atlas/trino/cli/ExtractorContext.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.atlas.trino.cli;
+
+import org.apache.atlas.ApplicationProperties;
+import org.apache.atlas.AtlasException;
+import org.apache.atlas.trino.client.AtlasClientHelper;
+import org.apache.atlas.trino.client.TrinoClientHelper;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.configuration.Configuration;
+
+import java.io.IOException;
+
+public class ExtractorContext {
+    static final String TRINO_NAMESPACE_CONF         = "atlas.trino.namespace";
+    static final String DEFAULT_TRINO_NAMESPACE      = "cm";
+    static final String OPTION_CATALOG_SHORT         = "c";
+    static final String OPTION_CATALOG_LONG          = "catalog";
+    static final String OPTION_SCHEMA_SHORT          = "s";
+    static final String OPTION_SCHEMA_LONG           = "schema";
+    static final String OPTION_TABLE_SHORT           = "t";
+    static final String OPTION_TABLE_LONG            = "table";
+    static final String OPTION_CRON_EXPRESSION_SHORT = "cx";
+    static final String OPTION_CRON_EXPRESSION_LONG  = "cronExpression";
+    static final String OPTION_HELP_SHORT            = "h";
+    static final String OPTION_HELP_LONG             = "help";
+
+    private final Configuration     atlasConf;
+    private final String            namespace;
+    private final String            catalog;
+    private final String            schema;
+    private final String            table;
+    private final AtlasClientHelper atlasClientHelper;
+    private final TrinoClientHelper trinoClientHelper;
+    private final String            cronExpression;
+
+    public ExtractorContext(CommandLine cmd) throws AtlasException, IOException {
+        this.atlasConf         = getAtlasProperties();
+        this.atlasClientHelper = createAtlasClientHelper();
+        this.trinoClientHelper = createTrinoClientHelper();
+        this.namespace         = atlasConf.getString(TRINO_NAMESPACE_CONF, DEFAULT_TRINO_NAMESPACE);
+        this.catalog           = cmd.getOptionValue(OPTION_CATALOG_SHORT);
```

Comment:
I think it should be possible to set all configurations (catalog, schema, table, cronExpression) in atlas-trino-extractor-properties file. It should be possible to overrider them via command-line arguments.

```
atlas.trino.extractor.catalog=
atlas.trino.extractor.schema=
atlas.trino.extractor.table=
atlas.trino.extractor.schedule=
```